Hw11

EmelyanovPavel opened the pull request:
Написаны коды по заданиям из файла пдф.

Commit:
Написаны коды с использованием стл, векторов и т.д.

EmelyanovPavel committed:
Update Arrays2.cpp

Переведен комментарий на английский язык в теле оператора иф из цикла фор, написанного во второй функции войд.

tak333 commented on Arrays2.cpp:
```diff
@@ -0,0 +1,73 @@
+// //12. Count the number of maximum elements.
+
+void task1() {
+    int size, k;
```
Объявляем переменные по мере необходимости.

tak333 commented on Arrays2.cpp:
```diff
@@ -0,0 +1,73 @@
+    std::cout << "Enter an array size:\n";
+    std::cin >> size;
+
+    int* arr = new int[size];
```
Работаем с векторами.

tak333 commented on Arrays2.cpp:
```diff
@@ -0,0 +1,73 @@
+    }
+
+    auto max = std::max_element(arr, arr + size);
+    k = std::count_if(arr, arr + size, [&](int i) { return i == *max; });
```
Захватываем только то, что нужно.

tak333 commented on Arrays2.cpp:
```diff
@@ -0,0 +1,73 @@
+    std::vector<int> result;
+
+    // Using the iterator to iterate through the elements
+    for (size_t i = 0; i < numbers.size(); i++) {
```
Используем средства algorithm для этого.

tak333 replied:
Вам нужно менять старый вектор, а не создавать новый. Работайте с итераторами.

EmelyanovPavel committed:
Update Arrays2.cpp

Написаны коды с векторами в функциях void и добавлены алгоритмы в первую функцию void.

EmelyanovPavel committed:
Update Arrays2.cpp

Написан код по заданию №2 с использованием итераторов и векторов, и без создания второго вектора.
